JavaScript/es6Guidelines.md

```diff
@@ -0,0 +1,189 @@
+We listed some useful tips to help you decide which declaration to use.
+
+1.  **Are you going to reassign the variable's value?**  
+    **no)** use const  
+    **yes)** use let
+
+    _"No reassign" means - you are not going to change the reference._  
+
+    **Something like that is correct:**
+
+    ```javascript
+    const point = {
+      x: 5,
+      y: 5
+    };
+    point.x = 10
+    ```
+
+   **However, something like this is not:**
+
+    ```javascript
+    const shape = 'triangle';
+    shape = 'circle';
+    ```
+
+2.  **Always use one let/const/var declaration in block**  
+    You should put lets and consts on top of the block.  
+    Even though consts and lets are not hoisted like vars it's a good practice to put those declarations on top of the block to increase code readability.
+
+3.  **Try to keep let/const/var scope as small as possible**
+
+4.  **When to use `var`?**
+
+    **Never**. No, seriously. There are almost no valid use cases for `var` right now.
+
+### Arrow functions gotchas
+1.  **Lexical `this` Binding**
+
+    `this` is inherited from the scope of which the arrow function is defined
+
+2.  **`this` cannot be changed**
+
+    this cannot be changed inside the function
+
+3.  **Implied Returns**
+
+    Curly braces are optional for arrow functions, they can be left off to create a super simple function. In those types of definitions, the result of the expression is always returned.
+
+    Example:
+
+    ```javascript
+    let sum = (x, y) => x + y;
+    ```
+
+
+4.  **`arguments` gotcha**
+
+    Check out following example: (code comes from [Ooyala Player](https://github.com/Wikia/mercury/blob/dev/front/scripts/mercury/modules/VideoPlayers/Ooyala.js))
+    ```javascript
+    setupPlayer() {
+      this.params = $.extend(this.params, {
+        onCreate: (...args) => {
+          return this.onCreate.apply(this, args);
+        }
+      });
+
+      if (!window.OO) {
+        this.loadPlayer();
+      } else {
+        this.createPlayer();
+      }
+    }
+    ```
+
+    Note the usage of `...args` instead of `arguments` keyword. **Why?**
+
+    Well, we used `arguments` and it turned out that **arrow functions don't have lexical `arguments`**. So instead `arguments` from setupPlayer were used. What we really wanted was to use `arguments` passed to onCreate hook called by Ooyala, hence usage of [rest parameter](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/rest_parameters).
+
+Recommended read:
+* http://blog.getify.com/arrow-this/
+* http://toddmotto.com/es6-arrow-functions-syntaxes-and-lexical-scoping/
+* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions
+
+### Default Exports
+
+**Always add export default**
+
+When you don't know which property you should export as default a one think about the property which is the best representative of particular module or which one is going to be used externally most often.

```

Comment:
IMO this would be easier to read:
>If you don't know which property should be exported as the default, decide which one is the best representative of particular module or which one is going to be used externally the most frequently.